src/Moq/ISetup.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System.Reflection;
+
+namespace Moq
+{
+	internal interface ISetup
```

Comment:
I really dislike having this interface. We don't really want two parallel type hierarchies (`Setup` and `ISetup`). It might be better, after all, to just let `AutoImplementedPropertyGetterSetup` sub-class `FixedReturnValueSetup`, which would have to be renamed to e.g. `DeterministicReturnValueSetup`. This would incur a minute additional overhead in `AutoImplementedPropertyGetterSetup`; but we could get rid of these parallel interfaces.